CustomerSettingsActivity.java

```diff
@@ -0,0 +1,185 @@
+package com.simcoder.uber;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomerSettingsActivity extends AppCompatActivity {
+
+    private EditText mNameField, mPhoneField;
+
+    private Button mBack, mConfirm;
+
+    private ImageView mProfileImage;
+
+    private FirebaseAuth mAuth;
+    private DatabaseReference mCustomerDatabase;
+
+    private String userID;
+    private String mName;
+    private String mPhone;
+    private String mProfileImageUrl;
+
+    private Uri resultUri;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_customer_settings);
+
+        mNameField = (EditText) findViewById(R.id.name);
+        mPhoneField = (EditText) findViewById(R.id.phone);
+
+        mProfileImage = (ImageView) findViewById(R.id.profileImage);
+
+        mBack = (Button) findViewById(R.id.back);
+        mConfirm = (Button) findViewById(R.id.confirm);
+
+        mAuth = FirebaseAuth.getInstance();
+        userID = mAuth.getCurrentUser().getUid();
+        mCustomerDatabase = FirebaseDatabase.getInstance().getReference().child("Users").child("Customers").child(userID);

```

Comment:
When method calls are chained, vertical composition will increase readability:
```
FirebaseDatabase.getInstance()
      .getReference()
      .child("Users")
      .child("Customers")
      .child(userID);
```
this could also be in a separate method:
```
private DatabaseReference resolveCustomerDatabase() {
    return FirebaseDatabase.getInstance()
        .getReference()
        .child("Users")
        .child("Customers")
        .child(userID);
}
```